x/metadata/client/cli/tx.go

```diff
@@ -1264,19 +1302,31 @@ func validateAccAddress(addr, argName string) (string, error) {
 	return addr, nil
 }
 
-// ParseNetAssetValueString splits string (example 1hotdog,2jackthecat) to list of NetAssetValue's
+// ParseNetAssetValueString splits string (example 1hotdog,1;2jackthecat100,...) to list of NetAssetValue's
 func ParseNetAssetValueString(netAssetValuesString string) ([]types.NetAssetValue, error) {
-	navs := strings.Split(netAssetValuesString, ",")
+	navs := strings.Split(netAssetValuesString, ";")
 	if len(navs) == 1 && len(navs[0]) == 0 {
 		return []types.NetAssetValue{}, nil
 	}
 	netAssetValues := make([]types.NetAssetValue, len(navs))
 	for i, nav := range navs {
-		coin, err := sdk.ParseCoinNormalized(nav)
+		parts := strings.Split(nav, ",")
+		if len(parts) != 1 && len(parts) != 2 {
+			return []types.NetAssetValue{}, errors.New("invalid net asset value, expected [coin,volume] or [coin]")
+		}
+		coin, err := sdk.ParseCoinNormalized(parts[0])
 		if err != nil {
-			return []types.NetAssetValue{}, fmt.Errorf("invalid net asset value coin : %s", nav)
+			return []types.NetAssetValue{}, fmt.Errorf("invalid net asset value coin : %s", parts[0])
+		}
+		if len(parts) == 2 {
+			volume, err := strconv.ParseUint(parts[1], 10, 64)
+			if err != nil {
+				return []types.NetAssetValue{}, fmt.Errorf("invalid volume : %s", parts[1])
+			}
+			netAssetValues[i] = types.NewNetAssetValue(coin, volume)
+		} else {
+			netAssetValues[i] = types.NewNetAssetValue(coin, 1)
 		}
-		netAssetValues[i] = types.NewNetAssetValue(coin)
 	}
 	return netAssetValues, nil
```

Comment:
Use of a Decimal in this location is completely against the entire design of forcing these values to be integers and remaining a ratio.  A volume should be greater than zero however to avoid an irrational result.